### PR TITLE
OCPBUGS-43048: aws/capi: set CIDR block for Node Port Service ingress rule

### PR DESCRIPTION
**Pull request by r4f4**
~Upstream capa is adding an extra Node Port Service ingress rule allowing any IPs. We already add our own rule restricted to the cluster's node security groups. While we work with upstream, let's remove this extra rule in our bootstrap post-destroy stage.~

Edit: with https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/5147 we can set a custom CIDR block for the rule, so we don't need to remove the rule anymore as a workaround.

### CODE REVIEW

File: pkg/infrastructure/aws/clusterapi/aws.go [hunk 1]
```diff
@@ -495,3 +510,45 @@ func keys[K comparable, V any](m map[K]V) []K {
...
 	return s
 }
+
+// removeNodePortServiceAllRule removes a too-permissive capa-upstream ingress rule.
+func removeNodePortServiceAllRule(ctx context.Context, session *session.Session, sgID string) error {
+	input := &ec2.RevokeSecurityGroupIngressInput{
+		GroupId: aws.String(sgID),
+		IpPermissions: []*ec2.IpPermission{
+			{
+				IpProtocol: aws.String("tcp"),
```
[the remark this answers is not in the document]
Comment (r4f4) [reply pkg/infrastructure/aws/clusterapi/aws.go:520]: These are the installer-created rules. They are restricted to the controlPlane/node security group rules. The rule I was addressing is the CAPA-created one [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/pkg/cloud/services/securitygroup/securitygroups.go#L647-L658).

But thanks to your question, I've just realized upstream [added a way to override that cidr block](https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/5147) last week so we can not even create this rule in the first place!

**Comment by r4f4** [comment]
/hold

There is a better solution with upstream support. I'll update the PR.

**Comment by r4f4** [comment]
/hold cancel

This is now using the upstream fix and in a local deploy I see the rule source as "10.0.0.0/16" instead of "0.0.0.0/0"

**Comment by r4f4** [comment]
/retitle OCPBUGS-43048: aws/capi: set CIDR block for Node Port Service ingress rule

**Comment by patrickdillon** [comment]
This LGTM. If we can't backport the newer version of CAPA, will we just use the downstream solution?

**Comment by r4f4** [comment]
> This LGTM. If we can't backport the newer version of CAPA, will we just use the downstream solution?

We can backport to 4.17 since it's a minor bump for capi from 1.8.2 to 1.8.4. But it will be risky for 4.16 with capi 1.7.0, so we might have to go with the workaround in that release.

**Comment by r4f4** [comment]
Update: dropped the capi bump as it's already been done by https://github.com/openshift/installer/pull/9118

**Review by barbacbd — COMMENTED** [review]
/lgtm 


note: https://github.com/openshift/installer/pull/9127#issuecomment-2438359952

**Pull request merged**